internal/cache/controller_obj_cache.go

```diff
@@ -2763,14 +2802,73 @@ func checkAndSetCloudType(client *clients.AviClient) bool {
 	return true
 }
 
+func checkIPAMForUsableNetworkLabels(client *clients.AviClient, ipamRefUri *string) bool {
+	// Donot check for labels in usable networks if a vipNetwork is provided by the user.
+	// In this case, the vipNetwork provided by the user will be used.
+	// 1. Prioritize user input vipetworkList, skip marker based selection if provided.
+	// 2. If not provided, check for markers in ipam's usable networks.
+	// 3. If marker based usable network is not available, keep vipNetworkList empty.
+	// 4. vipNetworkList can be empty only in WCP usecases, for all others, mark invalid configuration.
+
+	// 1. User input
+	if vipList, _ := lib.GetVipNetworkListEnv(); len(vipList) > 0 {
+		lib.SetVipNetworkList(vipList)
+		return true
+	}
+
+	// 2. Marker based, not valid for public clouds.
+	if ipamRefUri != nil {
+		clusterName := lib.GetClusterName()
+		if lib.GetAdvancedL4() {
+			clusterName = lib.GetClusterID()
+		}
+
+		ipam := models.IPAMDNSProviderProfile{}
+		ipamRef := strings.SplitAfter(*ipamRefUri, "/api/")
+		if err := lib.AviGet(client, "/api/"+ipamRef[1], &ipam); err != nil {
+			utils.AviLog.Errorf("Get uri %v returned err %v", ipamRef, err)
+			return false
+		}
+
+		markerNetworkFound := ""
+		for _, usableNetwork := range ipam.InternalProfile.UsableNetworks {
+			networkRef := strings.SplitAfter(*usableNetwork.NwRef, "/api/")
+			network := models.Network{}
+			if err := lib.AviGet(client, "/api/"+networkRef[1], &network); err != nil {
+				utils.AviLog.Errorf("Get uri %v returned err %v", networkRef, err)
+				return false
+			}
+
+			if len(network.Markers) == 1 &&
+				*network.Markers[0].Key == lib.ClusterNameLabelKey &&
+				len(network.Markers[0].Values) == 1 &&
+				network.Markers[0].Values[0] == clusterName {
+				markerNetworkFound = *network.Name
+			}
+		}
+
+		if markerNetworkFound != "" {
+			lib.SetVipNetworkList([]akov1alpha1.AviInfraSettingVipNetwork{{
+				NetworkName: markerNetworkFound,
+			}})
```

Comment:
Why wouldn't you return here? It looks like you will re-execute the code block in line 2858 even if you found a marker labelled network?